### PR TITLE
Added support to choose from only valid combinations of database type…

### PR DESCRIPTION
**Pull request by yash0024**
…, source and authentication when creating a relational connection

## Summary
- Updated the UI so that only valid combinations of Datasource and Authentication strategies are shown given the database type.

## How did you test this change?

- [ ] Test(s) added
- [ ] Manual testing (please provide screenshots/recordings)
- [x] No testing (please provide an explanation)

(tried to add a video/screenshot but the files don't process)

### CODE REVIEW

File: .yarnrc.yml [hunk 1]
```diff
@@ -29,3 +29,15 @@ plugins:
     spec: '@yarnpkg/plugin-interactive-tools'
 
 yarnPath: .yarn/releases/yarn-3.5.0.cjs
+
+npmRegistryServer: http://npmscopes.site.gs.com
```
Comment (xannem) [inline .yarnrc.yml:33]: You can remove these parts the `.yarnrc.yml` shouldn't need to be part of your commit and can be ignored 👍

File: packages/legend-application-studio/src/components/editor/editor-group/connection-editor/RelationalDatabaseConnectionEditor.tsx [hunk 2]
```diff
@@ -1589,11 +1589,21 @@ const RelationalConnectionGeneralEditor = observer(
...
+      if (connectionValueState.selectedValidAuthenticationStrategies[0]) {
+        connectionValueState.changeAuthenticationStrategy(
+          connectionValueState.selectedValidAuthenticationStrategies[0],
+        );
+      }
     };
 
     // source spec type
     const sourceSpecOptions = (
-      Object.values(CORE_DATASOURCE_SPEC_TYPE) as string[]
```
Comment (xannem) [inline packages/legend-application-studio/src/components/editor/editor-group/connection-editor/RelationalDatabaseConnectionEditor.tsx:1596]: as discussed offline, remove unused imports of `CORE_DATASOURCE_SPEC_TYPE` and `CORE_AUTHENTICATION_STRATEGY_TYPE`

File: packages/legend-application-studio/src/stores/graph-modifier/STO_Relational_GraphModifierHelper.ts [hunk 3]
```diff
@@ -102,6 +102,12 @@ export const relationDbConnection_setNewAuthenticationStrategy = action(
   },
 );
 
+export const relationDbConnection_setDatasourceSpecificationFromType = action(
+  (con: RelationalDatabaseConnection): void => {
+    con;
```
Comment (xannem) [inline packages/legend-application-studio/src/stores/graph-modifier/STO_Relational_GraphModifierHelper.ts:107]: Are you doing something with this method?

**Review by xannem — CHANGES_REQUESTED** [review]
+comments

**Review by xannem — APPROVED** [review]

File: .yarnrc.yml [hunk 4]
```diff
@@ -28,4 +28,4 @@ plugins:
   - path: .yarn/plugins/@yarnpkg/plugin-interactive-tools.cjs
     spec: '@yarnpkg/plugin-interactive-tools'
 
-yarnPath: .yarn/releases/yarn-3.5.0.cjs
+yarnPath: .yarn/releases/yarn-3.5.0.cjs
```
Comment (xannem) [inline .yarnrc.yml:31]:
```suggestion
yarnPath: .yarn/releases/yarn-3.5.0.cjs

```

**Comment by MauricioUyaguari** [comment]
major concern is for auth and datasource that are not part of core legend-studio. lets say we add an auth calleed 'myAuth' and It is valid for H2.

**Comment by MauricioUyaguari** [comment]
As discussed with @xannem 
 there are more database types that a user can select with extensions, 
so you will need to edit the extension mechnaism to take in the database type and only return datasource speficiations/auth strategies that match the database type.

File: packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/connection/ConnectionEditorState.ts [hunk 5]
```diff
@@ -227,6 +230,93 @@ export class RelationalDatabaseConnectionValueState extends ConnectionValueState
     return undefined;
   }
 
+  get selectedValidDatasources(): Array<CORE_DATASOURCE_SPEC_TYPE> {
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/connection/ConnectionEditorState.ts:233]: CORE_DATASOURCE_SPEC_TYPE[]

File: packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/connection/ConnectionEditorState.ts [hunk 6]
```diff
@@ -227,6 +230,93 @@ export class RelationalDatabaseConnectionValueState extends ConnectionValueState
...
+      case DatabaseType.Redshift: {
+        return [CORE_DATASOURCE_SPEC_TYPE.REDSHIFT];
+      }
+      default: {
+        return Object.values(CORE_DATASOURCE_SPEC_TYPE);
+      }
+    }
+  }
+
+  get selectedValidAuthenticationStrategies(): Array<CORE_AUTHENTICATION_STRATEGY_TYPE> {
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/connection/ConnectionEditorState.ts:274]: CORE_AUTHENTICATION_STRATEGY_TYPE[]

**Pull request closed without merging**